component.yaml

```diff
@@ -46,12 +46,33 @@ inputs:
   description: "Maximum failure percentage limit to interrupt workflow. It must be in the range 1-100."
   helpText:
 - key: "AC_TESTINIUM_TIMEOUT"
-  defaultValue: "6000"
+  defaultValue: "60"
   isRequired: true
   title: Timeout
-  description: Testinium plan timeout in seconds
+  description: "Testinium plan timeout in minutes."
+  helpText:
+- key: "AC_TESTINIUM_MAX_API_RETRY_COUNT"
+  defaultValue: "4"
+  isRequired: true
+  title: Testinium Maximum API Repetition Count
+  description: "Determine max repetition in case of Testinium platform congestion or API errors."
   helpText:
 outputs:
+- key: "AC_TESTINIUM_RESULT_FAILURE_SUMMARY"
+  title: "Total Number of Failures"
+  defaultValue: "AC_TESTINIUM_RESULT_FAILURE_SUMMARY"
+  description: "Total number of failures in test results."
+  helpText:
+- key: "AC_TESTINIUM_RESULT_ERROR_SUMMARY"
+  title: "Total Number of Errors"
+  description: "Total number of errors in test results."
+  defaultValue: "AC_TESTINIUM_RESULT_FAILURE_SUMMARY"
+  helpText:
+- key: "AC_TESTINIUM_RESULT_SUCCESS_SUMMARY"
+  title: "Total Number of Successes"
+  defaultValue: "AC_TESTINIUM_RESULT_FAILURE_SUMMARY"
```

Comment:
IMO, we have a problem in "defaultValue" of the outputs. `AC_TESTINIUM_RESULT_FAILURE_SUMMARY` is repeating for all keys.

![Screenshot from 2023-11-28 20-28-57](https://github.com/appcircleio/appcircle-testinium-component/assets/7355362/82f6e85e-03a1-4c15-afa8-03eecebeca9c)

Reply:
 @ftmzhrsckldr I have an extra note. It may be related to this; I'm not sure. :monocle_face: 

I checked your "success" pipeline [here](https://dev-my.appcircle.io/build/detail/5f9bbe36-149f-4c8e-b6e2-04435f276add?modal=/build/modal/Logs&profileId=5f9bbe36-149f-4c8e-b6e2-04435f276add&commitId=2c3ed416-8f5f-48da-b9aa-a145adaecbaa&buildId=11f444cb-cf60-43ad-a243-ebac47ca13df&scope=build&method=get&os=1).

```ruby
result_summary = data[:result_summary]
```
We see `result_summary` as below in build log.

```
Test result summary: {:ERROR=>1}
```

![testinium-report-summary](https://github.com/appcircleio/appcircle-testinium-component/assets/7355362/9e2fcb45-ced2-4ffc-bf38-3559853239bb)

That's OK. :white_check_mark: 

According to the implementation below ...

```ruby
      result_failure_summary = result_summary[:FAILURE] || 0
      result_error_summary = result_summary[:ERROR] || 0
      result_success_summary = result_summary[:SUCCESS] || 0
      ...
      ...

      open("#{$env_file_path}", 'a') { |f|
        f.puts "AC_TESTINIUM_RESULT_FAILURE_SUMMARY=#{result_failure_summary}"
        f.puts "AC_TESTINIUM_RESULT_ERROR_SUMMARY=#{result_error_summary}"
        f.puts "AC_TESTINIUM_RESULT_SUCCESS_SUMMARY=#{result_success_summary}"
      }
```

We should see the environment variables like this. :point_down: 

```
AC_TESTINIUM_RESULT_FAILURE_SUMMARY: 0
AC_TESTINIUM_RESULT_ERROR_SUMMARY: 1
AC_TESTINIUM_RESULT_SUCCESS_SUMMARY: 0
```
**But** actually we see the output below. :point_down: 

```
AC_TESTINIUM_RESULT_FAILURE_SUMMARY: 0
AC_TESTINIUM_RESULT_ERROR_SUMMARY: 
AC_TESTINIUM_RESULT_SUCCESS_SUMMARY:
```

![output-environment-variables](https://github.com/appcircleio/appcircle-testinium-component/assets/7355362/852c487e-105f-4543-a9db-1632356817f9)